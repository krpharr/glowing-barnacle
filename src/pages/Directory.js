import React from "react";
import SelectDepartment from "../components/SelectDepartment";
import SelectSort from "../components/SelectSort";
import EmployeeList from "../components/EmployeeList";
import data from "../data/data";

class Directory extends React.Component {
    state = {
        employees: data,
        departments: [],
        department: "All",
        sort: "lastName",
        currentList: []
    };

    componentDidMount() {
        const departments = [...new Set(data.map(emp => emp.department))];
        departments.unshift("All");
        let currentList = this.sort(data, "lastName");
        this.setState({departments: departments, currentList: currentList});    
    }

    sort = (data, column) => {
        let sorted = data.slice();
        sorted.sort((a,b)=>{
            if(a[column] > b[column])return 1;
            if(a[column] < b[column])return -1;
            return 0;
        });
        return sorted;
    };

    filter = (department) => {
        let filtered = data.filter(emp => {
            return emp.department === department;
        });
        return filtered;
    };

    handleDepartmentChange = event => {
        console.log("handleDepartmentChange: ", event.target.value);
        let empByDept = (event.target.value === "All" ? data : this.filter(event.target.value));
        console.log(empByDept);
        empByDept = this.sort(empByDept, this.state.sort);
        this.setState({department: event.target.value, currentList: empByDept});
    };

    handleSortChange = event => {
        console.log("handleSortChange: ", event.target.value);
        let sorted = this.sort(this.state.currentList, event.target.value);
        console.log(sorted);
        this.setState({sort: event.target.value, currentList: sorted});
    };
    
    render(){
        return (
            <div>
                <h1>Department</h1>
                <SelectDepartment 
                    departments={this.state.departments} 
                    handleDepartmentChange={this.handleDepartmentChange} 
                    department={this.state.department}
                />
                <h2>Sort</h2>
                <SelectSort handleSortChange={this.handleSortChange} />
                <EmployeeList currentList={this.state.currentList} />
            </div>
        );
    }
}

export default Directory;
