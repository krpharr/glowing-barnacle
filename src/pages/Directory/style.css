.select-sort {
  text-align: right;
}

.sort {
  font-size: 10px;
  font-weight: 700;
}