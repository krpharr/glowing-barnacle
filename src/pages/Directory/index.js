import React from "react";
import SelectDepartment from "../../components/SelectDepartment";
import SelectSort from "../../components/SelectSort";
import EmployeeList from "../../components/EmployeeList";
import data from "../../data/data";
import "./style.css";

class Directory extends React.Component {
  state = {
    employees: data,
    departments: [],
    department: "All",
    sort: "lastName",
    currentList: []
  };

  componentDidMount() {
    const departments = [...new Set(data.map(emp => emp.department))];
    departments.unshift("All");
    let currentList = this.sort(data, "lastName");
    this.setState({departments: departments, currentList: currentList});    
  }

  sort = (data, column) => {
    let sorted = data.slice();
    sorted.sort((a,b)=>{
      if(a[column] > b[column])return 1;
      if(a[column] < b[column])return -1;
      return 0;
    });
    return sorted;
  };

  filter = (department) => {
    let filtered = data.filter(emp => {
      return emp.department === department;
    });
    return filtered;
  };

  handleDepartmentChange = event => {
    console.log("handleDepartmentChange: ", event.target.value);
    let empByDept = (event.target.value === "All" ? data : this.filter(event.target.value));
    console.log(empByDept);
    empByDept = this.sort(empByDept, this.state.sort);
    this.setState({department: event.target.value, currentList: empByDept});
  };

  handleSortChange = event => {
    console.log("handleSortChange: ", event.target.value);
    let sorted = this.sort(this.state.currentList, event.target.value);
    console.log(sorted);
    this.setState({sort: event.target.value, currentList: sorted});
  };

  handleClickCard = id => {
    console.log("handleClickCard: ", id);
    window.location = `/employee/${id}`;
  };
  
  render(){
    return (
      <div className="container">
        <h1 className="text-center">Sort Schmort, Inc.</h1>
        <div className="row mt-2">
          <div className="col-5">
            <h5>Department:</h5>
          </div>
          <div className="col-7">
            <SelectDepartment 
                departments={this.state.departments} 
                handleDepartmentChange={this.handleDepartmentChange} 
                department={this.state.department}
            />
          </div>                
        </div>
        <div className="row">
          <div className="col-12">
            <div className="select-sort">
              <div className="sort">Sort</div>
              <SelectSort department={this.state.department} handleSortChange={this.handleSortChange} />
            </div>                        
          </div>
        </div>
        <EmployeeList currentList={this.state.currentList} handleClickCard={this.handleClickCard}/>
      </div>
    );
  }
}

export default Directory;
