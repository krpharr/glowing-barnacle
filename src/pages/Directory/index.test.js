import React from "react";
import renderer from "react-test-renderer";
import Directory from "./index";

test("Directory snapshot test", () => {
  const component = renderer.create(<Directory />);
  const tree = component.toJSON;
  expect(tree).toMatchSnapshot();
});

