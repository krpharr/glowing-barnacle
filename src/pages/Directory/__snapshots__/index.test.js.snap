// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory snapshot test 1`] = `[Function]`;
